<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>850</width>
    <height>650</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>700</width>
    <height>500</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Lightweight Mouse Control</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="recordingGroupBox">
      <property name="title">
       <string>Recording Controls</string>
      </property>
      <layout class="QHBoxLayout" name="recordingLayout">
       <item>
        <widget class="QPushButton" name="recordButton">
         <property name="text">
          <string>Start Recording (Ctrl+B)</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="recordingStatusLabel">
         <property name="text">
          <string>Ready to record</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignmentFlag::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="playbackGroupBox">
      <property name="title">
       <string>Playback Controls</string>
      </property>
      <layout class="QHBoxLayout" name="playbackLayout">
       <item>
        <widget class="QPushButton" name="playButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Play Selected Path</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stopButton">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="speedLabel">
         <property name="text">
          <string>Speed:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="speedSpinBox">
         <property name="suffix">
          <string>x</string>
         </property>
         <property name="minimum">
          <double>0.100000000000000</double>
         </property>
         <property name="maximum">
          <double>5.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Orientation::Horizontal</enum>
      </property>
      <widget class="QGroupBox" name="pathListGroupBox">
       <property name="title">
        <string>Saved Paths</string>
       </property>
       <layout class="QVBoxLayout" name="pathListLayout">
        <item>
         <widget class="QListWidget" name="pathListWidget">
          <property name="selectionMode">
           <enum>QAbstractItemView::SelectionMode::ExtendedSelection</enum>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="pathButtonsLayout">
          <item>
           <widget class="QPushButton" name="renamePathButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Rename</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="deletePathButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Delete</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="batchDeleteButton">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Delete Selected</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="selectAllButton">
            <property name="text">
             <string>Select All</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="refreshPathsButton">
            <property name="text">
             <string>Refresh</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="pathDetailsGroupBox">
       <property name="title">
        <string>Path Details</string>
       </property>
       <layout class="QVBoxLayout" name="pathDetailsLayout">
        <item>
         <widget class="QTextEdit" name="pathDetailsTextEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
          <property name="placeholderText">
           <string>Select a path to view details...</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="bottomButtonsLayout">
      <item>
       <widget class="QPushButton" name="settingsButton">
        <property name="text">
         <string>Settings</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QLabel" name="intervalLabel">
        <property name="text">
         <string>Recording Interval: 50ms</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="hotkeyLabel">
        <property name="text">
         <string>Global Hotkey: Ctrl+B</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>850</width>
     <height>18</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionDiagnostics"/>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionDiagnostics">
   <property name="text">
    <string>Diagnostics</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
